libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2116,26 +2131,6 @@ bool TypeChecker::visit(MemberAccess const& _memberAccess)
 			annotation.isLValue = annotation.referencedDeclaration->isLValue();
 	}
 
-	if (exprType->category() == Type::Category::Contract)
```

Comment:
Why was this block removed?

Reply:
This whole block is covered by the case starting in line 2067, where we check for contracts whether a member that now only lives in ``Address`` was requested. So removing this whole block is fine!